### ISSUE TITLE
does not dump maps of beans correctly

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Mause/snakeyaml#3]
```
What steps will reproduce the problem?

import java.io.*;
import java.util.*;
import junit.framework.TestCase;
import org.yaml.snakeyaml.Yaml;

public class MapYamlTest extends TestCase {

    public void testYamlMap() throws IOException {
        Map<String, Bean> data = new TreeMap<String, Bean>();
        data.put("gold1", new Bean());
        data.put("gold2", new Bean());

        Yaml yaml = new Yaml();
        File file = new File("beantest.yml");

        FileOutputStream os = new FileOutputStream(file);
        yaml.dump(data, new OutputStreamWriter(os));
        os.close();

        FileInputStream is = new FileInputStream(file);
        Object o = yaml.load(new InputStreamReader(is));
        is.close();

        assertTrue(o instanceof Map);
        Map m = (Map)o;
        assertTrue(m.get("gold1") instanceof Bean);
        assertTrue(m.get("gold2") instanceof Bean);
    }

    public static class Bean {
        private String a;
        public Bean() { a = ""; }
        public String getA() { return a; }
        public void setA(String s) { a = s; }

    }

}


What is the expected output? What do you see instead?

Test should pass. However, it fails on the last assert. If you look at
beantest.yml it has this:

gold1: !![package omitted]MapYamlTest$Bean {a: ''}
gold2: {a: ''}

In other words, it's only writing the tag for the first entry in the map.


What version of the product are you using? On what operating system?

Latest SnakeYAML from the hg repo, on Debian


Please provide any additional information below.

N/A

```

Original issue reported on code.google.com by `infinity0x@gmail.com` on 2 Jul 2009 at 4:54
